unit/memleak_hawaii3.py

```diff
@@ -39,13 +39,23 @@
     ax.pcolor(10 * rand(50, 50))
 
     fig.savefig('tmp%d' % i, dpi=75)
-    close(1)
+    plt.close(1)
 
-    gc.collect()
+    coll = gc.collect()
     val = report_memory(i)
     print(i, val)
     if i == indStart:
         start = val  # wait a few cycles for memory usage to stabilize
+    mem_size.append(val)
+    coll_count.append(coll)
 
 end = val
-print('Average memory consumed per loop: %1.4fk bytes\n' % ((end - start) / float(indEnd - indStart)))
+print('Average memory consumed per loop: %1.4fk bytes\n' %
+      ((end - start) / float(indEnd - indStart)))
+fig, ax = plt.subplots()
+ax2 = ax.twinx()
+ax.plot(mem_size, 'r')
+ax.set_ylabel('memory size', color='r')
+ax2.plot(coll_count, 'k')
+ax2.set_ylabel('collect count', color='k')
+fig.savefig('report')
```

Comment:
Probably matters by your definition of "better". Better in keeping memory
under control while not causing much overhead, or better in keeping memory
usage to its smallest amounts?

On Thu, Oct 29, 2015 at 4:40 PM, Thomas A Caswell <notifications@github.com>
wrote:

> In unit/memleak_hawaii3.py
> <https://github.com/matplotlib/matplotlib/pull/5353#discussion_r43441186>:
>
> >
> >  end = val
> > -print('Average memory consumed per loop: %1.4fk bytes\n' % ((end - start) / float(indEnd - indStart)))
> > +print('Average memory consumed per loop: %1.4fk bytes\n' %
> > +      ((end - start) / float(indEnd - indStart)))
> > +fig, ax = plt.subplots()
> > +ax2 = ax.twinx()
> > +ax.plot(mem_size, 'r')
> > +ax.set_ylabel('memory size', color='r')
> > +ax2.plot(coll_count, 'k')
> > +ax2.set_ylabel('collect count', color='k')
> > +fig.savefig('report')
>
> I find the python version dependence strange, I thought the gc got
> *better* in 3...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/5353/files#r43441186>.
>